sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java

```diff
@@ -145,38 +149,55 @@ public ConfigurationClient buildClient() {
      * @throws IllegalStateException If {@link #connectionString(String) connectionString} has not been set.
      */
     public ConfigurationAsyncClient buildAsyncClient() {
+        // Global Env configuration store
         Configuration buildConfiguration =
             (configuration == null) ? Configuration.getGlobalConfiguration().clone() : configuration;
-        ConfigurationClientCredentials configurationCredentials = getConfigurationCredentials(buildConfiguration);
-        String buildEndpoint = getBuildEndpoint(configurationCredentials);
 
+        // Service version
+        ConfigurationServiceVersion serviceVersion =
+            version != null ? version : ConfigurationServiceVersion.getLatest();
+
+        // AAD token credential
+        TokenCredential tokenCredentialEnvVar = tokenCredential;
+        // Connection string credential
+        ConfigurationClientCredentials configurationCredentials = null;
+
+        // Endpoint
+        String buildEndpoint = endpoint;
+        if (tokenCredentialEnvVar == null) {
+            configurationCredentials = getConfigurationCredentials(buildConfiguration);
+            buildEndpoint = getBuildEndpoint(configurationCredentials);
+        }
         Objects.requireNonNull(buildEndpoint);
-        ConfigurationServiceVersion serviceVersion = version != null
-            ? version : ConfigurationServiceVersion.getLatest();
 
+        // Pipeline
         if (pipeline != null) {
             return new ConfigurationAsyncClient(buildEndpoint, pipeline, serviceVersion);
         }
 
-        ConfigurationClientCredentials buildCredential = (credential == null) ? configurationCredentials : credential;
-        if (buildCredential == null) {
-            throw logger.logExceptionAsWarning(new IllegalStateException("'credential' is required."));
-        }
-
         // Closest to API goes first, closest to wire goes last.
         final List<HttpPipelinePolicy> policies = new ArrayList<>();
-
-        policies.add(
-            new UserAgentPolicy(clientName, clientVersion, buildConfiguration,
-                serviceVersion));
+        policies.add(new UserAgentPolicy(clientName, clientVersion, buildConfiguration, serviceVersion));
         policies.add(new RequestIdPolicy());
         policies.add(new AddHeadersPolicy(headers));
         policies.add(new AddDatePolicy());
-        policies.add(new ConfigurationCredentialsPolicy(buildCredential));
-        HttpPolicyProviders.addBeforeRetryPolicies(policies);
 
-        policies.add(retryPolicy == null ? new RetryPolicy() : retryPolicy);
+        // add connection string credential if the AAD is not exist
+        if (tokenCredentialEnvVar == null) {
+            ConfigurationClientCredentials buildCredential =
+                (credential == null) ? configurationCredentials : credential;
```

Comment:
This is a weird line break. For ternary operations, be consistent about adding before `(` and `)`.